interpreter/text/parser.mly

```diff
@@ -39,6 +39,37 @@ let ati i =
 let literal f s =
   try f s with Failure _ -> error s.at "constant out of range"
 
+let range_check i32 min max at =
+  let i = Int32.to_int i32 in
+  if i > max || i < min then error at "constant out of range" else i
+
+let int8_of_string s = range_check (I32.of_string s.it) (-128) 255 s.at
+let int16_of_string s = range_check (I32.of_string s.it) (-32768) 65535 s.at
+let int32_of_string s = I32.to_bits (literal (fun s -> I32.of_string s.it) s)
+let int64_of_string s = I64.to_bits (literal (fun s -> I64.of_string s.it) s)
+let f32_of_string s = F32.to_bits (literal (fun s -> F32.of_string s.it) s)
+let f64_of_string s = F64.to_bits (literal (fun s -> F64.of_string s.it) s)
+
+let simd_literal shape ss =
+  let open Bytes in
+  let b = create 16 in
+  (match shape with
```

Comment:
Likewise, the logic for constructing simd values from lanes will probably have to move to vector.ml, since you'll need it elsewhere.

Reply:
I moved the construction of Simd128 values into s128, the parser deals with validation, then calls the of_strings constructor to build the value.